src/mesh/boundary_info.C

```diff
@@ -1228,9 +1228,10 @@ bool BoundaryInfo::has_boundary_id(const Elem * const elem,
 
 void BoundaryInfo::boundary_ids (const Elem * const elem,
                                  const unsigned short int side,
-                                 std::vector<boundary_id_type> & vec_to_fill) const
+                                 std::vector<boundary_id_type> & vec_to_fill, const bool traverse) const
 {
   libmesh_assert(elem);
+  libmesh_assert(_children_on_boundary || elem->level() == 0);
```

Comment:
This is seriously overzealous.  People do get to query boundary ids on active elements.